cpp/log/cert.cc

```diff
@@ -883,7 +914,8 @@ Cert::Status Cert::IsValidNameConstrainedIntermediateCa() const {
 
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
-  if (HasBasicConstraintCATrue() != Cert::TRUE
+  StatusOr<bool> has_ca_constraint_status = HasBasicConstraintCATrue();
```

Comment:
Yeah, and for extra fun, any `enum` is also an integral type, so HAHAHAHA! :wink: 

This problem is fixed by using the C++11 `enum class`:

http://en.cppreference.com/w/cpp/language/enum